src/uploaders.rs

```diff
@@ -51,16 +71,18 @@ impl Uploader {
     pub fn crate_location(&self, crate_name: &str, version: &str) -> Option<String> {
         match *self {
             Uploader::S3 {
+                ref host,
                 ref bucket,
                 ref cdn,
                 ..
             } => {
                 let host = match *cdn {
                     Some(ref s) => s.clone(),
-                    None => bucket.host(),
+                    None => host.clone(),
                 };
+
                 let path = Uploader::crate_path(crate_name, version);
-                Some(format!("https://{}/{}", host, path))
+                Some(format!("https://{}/{}/{}", host, bucket, path))
```

Comment:
It looks like this would change the location of where crates are stored in production.  If this is necessary, it will need to be configurable.